packages/pds/src/api/com/atproto/admin/updateSubjectState.ts

```diff
@@ -0,0 +1,56 @@
+import { CID } from 'multiformats/cid'
+import { AtUri } from '@atproto/syntax'
+import { Server } from '../../../../lexicon'
+import AppContext from '../../../../context'
+import {
+  isRepoRef,
+  isRepoBlobRef,
+} from '../../../../lexicon/types/com/atproto/admin/defs'
+import { isMain as isStrongRef } from '../../../../lexicon/types/com/atproto/repo/strongRef'
+import { AuthRequiredError, InvalidRequestError } from '@atproto/xrpc-server'
+
+export default function (server: Server, ctx: AppContext) {
+  server.com.atproto.admin.updateSubjectState({
+    auth: ctx.roleVerifier,
+    handler: async ({ input, auth }) => {
+      const access = auth.credentials
+      // if less than admin access then cannot perform a takedown
```

Comment:
This comment might be out-of-date since this check below permits non-admin moderators. Or maybe we need a more stringent check within the `if (takedown) {}` block below.

Reply:
ah good call 👍 

i updated the error text but not the comment